src/js/components/NewTodo.js

```diff
@@ -14,19 +12,20 @@ export default class NewTodo extends Component {
         super(props);
         this.state = {
             isNewTodoFormDisplayed: false
-        }
+        };
     }
 
     toggleNewTodoForm() {
-        this.setState({ isNewTodoFormDisplayed: !this.state.isNewTodoFormDisplayed });
+        this.setState({
+            isNewTodoFormDisplayed: !this.state.isNewTodoFormDisplayed
+        });
     }
 
     addNewItem() {
-        console.log(document.getElementById('description-textarea').value);
         this.toggleNewTodoForm();
+        let description = document.getElementById('description-textarea');
         let newItem = {
-            id: this.props.todoItems.length,
-            description: document.getElementById('description-textarea').value
+            description: description ? description.value : 'unknown item'
         };
 
         return this.props.dispatch(addNewItem(newItem));
```

Comment:
Better would be connecting action in @conect decorator, second argument https://github.com/reactjs/react-redux/blob/master/docs/api.md#connectmapstatetoprops-mapdispatchtoprops-mergeprops-options is mapDispatchToProps. So You could got action in props and invoke simply this.props.addNewItem(newItem)